src/frontend/views/Github/GithubTemplate.tsx

```diff
@@ -0,0 +1,171 @@
+import cx from 'classnames';
+
+import { Prompt } from 'react-router-dom';
+
+import { FormEventHandler, MouseEventHandler, Fragment } from 'react';
+
+import * as flowStyles from '../../components/CredentialFlow/CredentialFlow.module.css';
+import * as styles from './Github.module.css';
+
+import { Spinner } from '../../components/Spinner/Spinner';
+import { usePreventNavigation } from '../../utilities/usePreventNavigation';
+import { Explainer } from '../../components/Explainer/Explainer';
+import { LinkBack } from '../../components/LinkBack/LinkBack';
+import { AttestationErrorUnknown } from '../../components/AttestationErrorUnknown/AttestationErrorUnknown';
+import { ExpiryDate } from '../../components/ExpiryDate/ExpiryDate';
+import { DetailedMessage } from '../../components/DetailedMessage/DetailedMessage';
+import { AttestationProcessAnchoring } from '../../components/AttestationProcessAnchoring/AttestationProcessAnchoring';
+import { AttestationProcessReady } from '../../components/AttestationProcessReady/AttestationProcessReady';
+import { SigningErrorClosed } from '../../components/SigningErrorClosed/SigningErrorClosed';
+
+import { GithubProfile } from './Github';
+
+export type AttestationStatus =
+  | 'none'
+  | 'urlReady'
+  | 'authorizing'
+  | 'authorized'
+  | 'attesting'
+  | 'ready'
+  | 'error';
+
+export type FlowError = 'unauthorized' | 'closed' | 'unknown';
+
+interface Props {
+  status: AttestationStatus;
+  processing: boolean;
+  handleSubmit: FormEventHandler;
+  handleBackup: MouseEventHandler;
+  handleTryAgainClick: MouseEventHandler;
+  authUrl?: string;
+  flowError?: FlowError;
+  profile?: GithubProfile;
+}
+export function GithubTemplate({
+  status,
+  processing,
+  handleSubmit,
+  handleBackup,
+  handleTryAgainClick,
+  authUrl,
+  flowError,
+  profile,
+}: Props): JSX.Element {
+  const preventNavigation = usePreventNavigation(
+    processing || ['authorizing', 'authorized', 'attesting'].includes(status),
+  );
+
+  return (
+    <section
+      className={cx(flowStyles.container, {
+        [flowStyles.processing]: processing,
+      })}
+      aria-busy={processing}
+    >
+      {processing && <Spinner />}
+
+      <h1 className={styles.heading}>Github Account Attestation</h1>
+
+      <Prompt
+        when={preventNavigation}
+        message="The Github attestation process has already started. Are you sure you want to leave?"
+      />
+
+      <Explainer>
+        After you sign into your Github account and give SocialKYC a one-time
```

Comment:
Is it a one-time permission still?

Reply:
Isn't it? I think it's even more of a one-time permission than Discord, since the access token can't be revoked.